File: php-poo/aulas/aula05/ContaBanco.php
<?php

class ContaBanco
{
    // ATRIBUTOS
    public $numConta;
    protected $tipo;
    private $dono;
    private $saldo;
    private $status;

    // METODO CONSTRUTOR
    public function __construct()
    {
        $this->setSaldo(0); // $this->saldo = 0;
        $this->getStatus(false);  // $this->status = false;

        echo "<br> Conta criada com sucesso! <br>"; // criado para testar se a conta foi criada com sucesso
    }

    // METODOS Getters & Setters (get e set)

    public function getNumConta()
    {
        return $this->numConta;
    }

    public function setNumConta($numConta): void
    {
        $this->numConta = $numConta;
    }

    public function getTipo()
    {
        return $this->tipo;
    }

    public function setTipo($tipo): void
    {
        $this->tipo = $tipo;
    }

    public function getDono()
    {
        return $this->dono;
    }

    public function setDono($dono): void
    {
        $this->dono = $dono;
    }

    public function getSaldo()
    {
        return $this->saldo;
    }

    public function setSaldo($saldo): void
    {
        $this->saldo = $saldo;
    }

    public function getStatus()
    {
        return $this->status;
    }

    public function setStatus($status): void
    {
        $this->status = $status;
    }


    // METODOS ESPECIAIS

    public function abrirConta($t){
        $this->setTipo($t);
        $this->setStatus(true);

        if ($t == "CC") {
            $this->setSaldo(50);
        }
        elseif ($t == "CP") {
            $this->setSaldo(150);
        }
    }

    public function fecharConta() {

        if ($this->getSaldo() > 0) {
            echo "<br> [ERRO] Conta de {$this->getDono()} com dinheiro, saldo de R$ {$this->getSaldo()}, não posso fechar a conta! <br>";
        }
        elseif ($this->getSaldo() < 0) {
            echo "<br> [ERRO] Conta de {$this->getDono()} com débito, saldo de R$ {$this->getSaldo()}, não posso fechar a conta! <br>";
        }
        else {
            $this->setStatus(false);
            echo "<br> Conta de {$this->getDono()} fechada com sucesso. <br>";
        }
    }

    public function depositar($v){

        if ($this->getStatus() == true) {
            $this->setSaldo($this->getSaldo() + $v);
            echo "<br> Depósito de R$ $v na conta de {$this->getDono()}.<br>";
        } else {
            echo "<br> [ERRO] Impossível depositar <br>";
        }
    }

    public function sacar($v){

        if ($this->getStatus() == true) {
            if ($this->getSaldo() >= $v) {
                $this->setSaldo($this->getSaldo() - $v);
                echo "<br> Saque de R$ $v autorizado na conta de {$this->getDono()}. <br>";
                echo "<br> {$this->getDono()} - saldo da conta - R$ {$this->getSaldo()}. <br>";
            } else {
                echo "<br> {$this->getDono()} - [ERRO] Saldo Insuficiente para saque! <br>";
            }
        } else {
            echo "<br> [ERRO] Impossível sacar, conta está fechada! <br>";
        }
    }

    public function pagarMensal(){

        if ($this->getTipo() == "CC") {
            $v = 12;
        }
        elseif ($this->getTipo() == "CP") {
            $v = 20;
        }

        //  if ($this->getStatus())
        if ($this->getStatus() == true) {
            $this->setSaldo($this->getSaldo() - $v);
            echo "<br> Mensalidade de R$ $v debitada na conta de {$this->getDono()} <br>";
        }
        else {
            echo "<br>[ERRO] Impossível débitar a mensalidade, a conta está fechada! </br>";
        }
    }
}

File: java-poo/projects/poo-aula05/src/aula05/ContaBanco.java
package aula05;

public class ContaBanco {
    // ATRIBUTOS
    public int numConta;
    protected String tipo;
    private String dono;
    private float saldo;
    private boolean status;

    // MÉTODO CONSTRUTOR
    public ContaBanco() {
        this.setSaldo(0);
        this.setStatus(false);
    }

    // MÉTODOS ESPECIAIS

    public void estadoAtual() {
        System.out.println("------------------------------------------------------------");
        System.out.println("Conta: " + this.getNumConta());
        System.out.println("Tipo: " + this.getTipo());
        System.out.println("Dono: " + this.getDono());
        System.out.println("Saldo: " + this.getSaldo());
        System.out.println("Status: " + this.getStatus());
        System.out.println("------------------------------------------------------------");
        System.out.println("\n");
    }

    public void abrirConta(String tipo){
        this.setTipo(tipo);
        this.setStatus(true);

        if ( tipo == "CC") {
            this.setSaldo(50);
        }
        else if ( tipo == "CP" ) {
            this.setSaldo(150);
        }
        System.out.println(this.getDono() + ". Conta aberta com sucesso!");
    }

    public void fecharConta(){
        if ( this.getSaldo() > 0 ) {
            System.out.println("[ERRO] Conta de " +
                    this.getDono() + " está com dinheiro, saldo de R$ " +
                    this.getSaldo() + ", não posso fechar a conta!");
        }
        else if ( this.getSaldo() < 0 ) {
            System.out.println("[ERRO] Conta de " +
                    this.getDono() + " está com débito, saldo de R$ " +
                    this.getSaldo() + ", não posso fechar a conta!");
        }
        // ( this.getSaldo == 0)
        else {
            setStatus(false);
            System.out.println("Conta de " + this.getDono() + " foi fechada com sucesso!");
        }
    }

    public void depositar(float v){
        if ( this.getStatus() ) {
            this.setSaldo(this.getSaldo() + v); // this.saldo = this.saldo + v;
            System.out.println("Depósito de R$ " + v + " na conta de " + this.getDono());
        } else {
            System.out.println("ERRO. Impossível depositar em uma conta fechada!");
        }
    }
    public void sacar(float v){

        if ( this.getStatus() ) {
            if ( this.getSaldo() >= v) {
                this.setSaldo(this.getSaldo() - v);
                System.out.println("Saque de R$ " + v + " na conta de " + this.getDono());
                System.out.println(this.getDono() + ", saldo da conta R$ " + this.getSaldo());
            } else {
                System.out.println(this.getDono() + ", saldo insuficiente para saque!");
            }
        } else {
            System.out.println("[ERRO] Impossível sacar de uma conta está fechada!");
        }
    }

    public void pagarMensalidade(){
        float v = 0.0f;

        if (this.getTipo() == "CC") {
            v = 12;
        } else if (this.getTipo() == "CP") {
            v = 20;
        }

        if ( this.getStatus() ) {
            if ( this.getSaldo() > v) {
                this.setSaldo(this.getSaldo() - v);
                System.out.println("Mensalidade de R$ " + v + " debitada na conta de " +
                        this.getDono() + ". Saldo disponível R$ " + this.getSaldo());

            } else {
                System.out.println("Saldo negativo");
            }

        } else {
            System.out.println("[ERRO] Impossível débitar a mensalidade, a conta está fechada!");
        }

    }


    // Métodos Getters e Setters
    public int getNumConta() {
        return numConta;
    }

    public void setNumConta(int numConta) {
        this.numConta = numConta;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getDono() {
        return dono;
    }

    public void setDono(String dono) {
        this.dono = dono;
    }

    public float getSaldo() {
        return saldo;
    }

    public void setSaldo(float saldo) {
        this.saldo = saldo;
    }

    public boolean getStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }
}
